frontend/src/reducer/notificationPage.js

```diff
@@ -280,12 +281,47 @@ export default function notificationPage (state = defaultNotificationsObject, ac
     }
 
     case `${READ}/${NOTIFICATION_LIST}`: {
-      const notificationList = state.list.map(notification => (
-        { ...notification, read: true }
-      ))
+      const notificationList = state.list.map(notification => notification.group
+        ? { ...notification, group: notification.group.map(notification => ({ ...notification, read: true })) }
+        : { ...notification, read: true }
+      )
       return { ...state, list: uniqBy(notificationList, 'id'), unreadMentionCount: 0, unreadNotificationCount: 0 }
     }
 
+    case `${READ}/${CONTENT}/${NOTIFICATION}`: {
+      let unreadMentionCount = state.unreadMentionCount
+      let unreadNotificationCount = state.unreadNotificationCount
+      const newNotificationList = state.list.map(notification => {
+        if (notification.group) {
+          return {
+            ...notification,
+            group: notification.group.map(notification => {
+              if (!notification.content) return notification
+              if (getMainContentId(notification) === action.contentId) {
+                if (!notification.read) {
+                  if (notification.type.includes(TLM_ET.MENTION)) unreadMentionCount--
+                  unreadNotificationCount--
```

Comment:
I think we should avoid modifying outer variables in a map call when possible, but I'm good with it.